src/gui/syncconflictsmodel.cpp

```diff
@@ -0,0 +1,355 @@
+    result[static_cast<int>(SyncConflictRoles::ConflictPreviewUrl)] = "conflictPreviewUrl";
+
+    return result;
+}
+
+Qt::ItemFlags SyncConflictsModel::flags(const QModelIndex &index) const
+{
+    auto result = Qt::ItemFlags{};
+
+    if (!index.parent().isValid()) {
+        result = QAbstractListModel::flags(index);
+        return result;
+    }
+
+    result = Qt::ItemIsSelectable | Qt::ItemIsEditable;
+    return result;
+}
+
+ActivityList SyncConflictsModel::conflictActivities() const
+{
+    return mData;
+}
+
+bool SyncConflictsModel::allExistingsSelected() const
+{
+    return mAllExistingsSelected;
+}
+
+bool SyncConflictsModel::allConflictingSelected() const
+{
+    return mAllConflictingsSelected;
+}
+
+void SyncConflictsModel::setConflictActivities(ActivityList conflicts)
+{
+    if (mData == conflicts) {
+        return;
+    }
+
+    beginResetModel();
+
+    mData = conflicts;
+    emit conflictActivitiesChanged();
+
+    updateConflictsData();
+
+    endResetModel();
+}
+
+void SyncConflictsModel::selectAllExisting(bool selected)
+{
+    for (auto &singleConflict : mConflictData) {
+        singleConflict.mExistingSelected = selected;
+    }
+
+    Q_EMIT dataChanged(index(0), index(rowCount() - 1), {static_cast<int>(SyncConflictRoles::ExistingSelected)});
+
+    if (selected && !mAllExistingsSelected) {
+        mAllExistingsSelected = true;
+        Q_EMIT allExistingsSelectedChanged();
+    } else if (!selected && mAllExistingsSelected) {
+        mAllExistingsSelected = false;
+        Q_EMIT allExistingsSelectedChanged();
+    }
+}
+
+void SyncConflictsModel::selectAllConflicting(bool selected)
+{
+    for (auto &singleConflict : mConflictData) {
+        singleConflict.mConflictSelected = selected;
+    }
+
+    Q_EMIT dataChanged(index(0), index(rowCount() - 1), {static_cast<int>(SyncConflictRoles::ConflictSelected)});
+
+    if (selected && !mAllConflictingsSelected) {
+        mAllConflictingsSelected = true;
+        Q_EMIT allConflictingSelectedChanged();
+    } else if (!selected && mAllConflictingsSelected) {
+        mAllConflictingsSelected = false;
+        Q_EMIT allConflictingSelectedChanged();
+    }
+}
+
+void SyncConflictsModel::applyResolution()
+{
+    for(const auto &syncConflict : qAsConst(mConflictData)) {
+        if (syncConflict.isValid()) {
+            qCInfo(lcSyncConflictsModel) << syncConflict.mExistingFilePath << syncConflict.mConflictingFilePath << syncConflict.solution();
+            ConflictSolver solver;
+            solver.setLocalVersionFilename(syncConflict.mConflictingFilePath);
+            solver.setRemoteVersionFilename(syncConflict.mExistingFilePath);
+            solver.exec(syncConflict.solution());
+        }
+    }
+}
+
+void SyncConflictsModel::updateConflictsData()
+{
+    mConflictData.clear();
+    mConflictData.reserve(mData.size());
+
+    for (const auto &oneConflict : qAsConst(mData)) {
+        if (!FolderMan::instance()) {
```

Comment:
@mgallien is this even possible?

Reply:
I am not sure
I removed it as it was not needed